src/Utilities/src/MoistThermodynamics.jl

```diff
@@ -411,10 +413,17 @@ The optional input value of the temperature `T_init` is taken as the initial
 value of the saturation adjustment iterations.
 """
 function saturation_adjustment(E_int, ρ, q_t, T_init = T_triple)
-
-    T = find_zero(x -> internal_energy_sat(x, ρ, q_t) - E_int, T_init,
-            Order1(), atol=1e-3*cv_d)
-
+    tol_abs = 1e-3*cv_d
+    iter_max = 100
+    args = ()
+    T_0, T_1 = T_init, T_init*1.01
```

Comment:
air_temperature(E_int, q_t, 0, 0) and air_temperature(E_int, q_t, 0, q_t) should bracket the zero and would make good initial values. (Please check.)

Reply:
I tried
`T_0 = air_temperature(E_int, q_t, 0.0, 0.0)`
`T_1 = air_temperature(E_int, q_t, 0.0, q_t)`
and I'm getting a domain error in the saturation adjustment. I tried printing these temperatures, and I'm getting 
`T_0 = -473.59...`
`T_1 = 200.04...`
It looks like there is something wrong with `air_temperature`.

Reply:
I realize now, that we're not using `T_init = T_triple`. If I replace `T_0` with `T_init`, the tests pass, but I don't know that IC will sufficiently bound the equilibrium solution in general.

Reply:
air_temperature should be ok. You can just get negative values if E_int is small and q_t is relatively large. For the initial values for saturation adjustment, you can just replace the T_0 call by `T_0 = max(0, air_temperature(E_int, q_t, 0.0, 0.0))`, to prevent T_0 from going below zero.

Reply:
Hmm, getting `NaN`'s with `T_0 = max(0.0, air_temperature(E_int, q_t, 0.0, 0.0))`. I'm going to look at the equations more closely over the weekend and see if I can figure this out.

Reply:
Please replace l. 338 in MoistThermodynamics by `return min(eltype(ρ)(1.), p_vs / (ρ * R_v * T));`. Some of the tests had extreme (very unrealistic) values for q_t and T, resulting in specific humidities > 1. This change will bound saturation specific humidities by 1.